mirar/database/engine.py

```diff
@@ -2,27 +2,49 @@
 Util functions for database interactions
 """
 
-from sqlalchemy import Engine, NullPool, create_engine
+from sqlalchemy import URL, Engine, NullPool, create_engine
 
-from mirar.database.credentials import DB_PASSWORD, DB_USER
+from mirar.database.credentials import (
+    DB_HOSTNAME,
+    DB_PASSWORD,
+    DB_PORT,
+    DB_SCHEMA,
+    DB_USER,
+)
 
 
 def get_engine(
     db_name: str,
     db_user: str = DB_USER,
     db_password: str = DB_PASSWORD,
+    db_hostname: str = DB_HOSTNAME,
+    db_port: int = DB_PORT,
+    db_schema: str = DB_SCHEMA,
 ) -> Engine:
     """
     Function to create a postgres engine
 
     :param db_user: User for db
     :param db_password: password for db
     :param db_name: name of db
+    :param db_hostname: hostname of db
+    :param db_port: port of db
+    :param db_schema: schema of db
     :return: sqlalchemy engine
     """
 
+    url_object = URL.create(
+        "postgresql+psycopg",
+        username=db_user,
+        password=db_password,
+        host=db_hostname,
+        port=db_port,
+        database=db_name,
+    )
+
     return create_engine(
-        f"postgresql+psycopg://{db_user}:{db_password}" f"@/{db_name}",
+        url_object,
         future=True,
         poolclass=NullPool,
+        connect_args={"options": f"-csearch_path={db_schema}"},
```

Comment:
What will happen hear if schema is None?

Reply:
If db_schema is None, it will try to search as if the schema name is None.
This is why the default value if not set for the DB_SCHEMA environment variable is "public" which is the default schema for postgres.